src/pages/api/graphql.tsx

```diff
@@ -62,36 +53,36 @@ const clientConnection = async () => {
   }
 }
 
-export const apolloServer = new ApolloServer({
+export const apolloServer = new ApolloServer<ServerContext>({
   typeDefs,
   resolvers,
-  dataSources: () => {
-    return {
-      keystoneAPI: new KeystoneAPI(),
-      weatherAPI: new WeatherAPI(),
-      personnelAPI: new PersonnelAPI(),
-    }
-  },
+  status400ForVariableCoercionErrors: true,
   cache: 'bounded',
   plugins: [ApolloServerPluginLandingPageDisabled()],
-  context: async ({ req, res }) => {
+})
+
+export default startServerAndCreateNextHandler(apolloServer, {
```

Comment:
This function is provided by the apollo server nextjs integration library we're using. Setup guide here: https://github.com/apollo-server-integrations/apollo-server-integration-next

Everything that was previously passed in as context to the server directly, is now passed in as an argument to this function.